src/ec/curve25519/ed25519.rs

```diff
@@ -202,6 +202,100 @@ impl<'a> Ed25519KeyPair {
     }
 }
 
+/// The components of an Ed25519 key pair. This type cannot be used
+/// for signing, see [Ed25519KeyPair](./Ed25519KeyPair.html) instead.
+pub struct Ed25519KeyPairComponents {
+    /// The 32-byte private part of this pair, i.e. the seed.
+    pub private_key: [u8; 32],
+    /// The 32-byte public part of this key.
+    pub public_key: [u8; 32],
+}
+
+impl<'a> Ed25519KeyPairComponents {
+    /// Constructs an Ed25519 key pair by parsing an unencrypted PKCS#8 v2
+    /// Ed25519 private key.
+    ///
+    /// The input must be in PKCS#8 v2 format, and in particular it must contain
+    /// the public key in addition to the private key. `from_pkcs8()` will
+    /// verify that the public key and the private key are consistent with each
+    /// other.
+    ///
+    /// If you need to parse PKCS#8 v1 files (without the public key) then use
+    /// `Ed25519KeyPair::from_pkcs8_maybe_unchecked()` instead.
+    pub fn from_pkcs8(input: untrusted::Input)
+                      -> Result<Ed25519KeyPairComponents, error::Unspecified> {
+        let (seed, public_key) = unwrap_pkcs8(pkcs8::Version::V2Only, input)?;
+        Self::from_seed_and_public_key(seed, public_key.unwrap())
+    }
+
+    /// Constructs an Ed25519 key pair from the private key seed `seed` and its
+    /// public key `public_key`.
+    ///
+    /// It is recommended to use `Ed25519KeyPair::from_pkcs8()` instead.
+    ///
+    /// The private and public keys will be verified to be consistent with each
+    /// other. This helps avoid misuse of the key (e.g. accidentally swapping
+    /// the private key and public key, or using the wrong private key for the
+    /// public key). This also detects any corruption of the public or private
+    /// key.
+    pub fn from_seed_and_public_key(seed: untrusted::Input,
+                                    public_key: untrusted::Input)
```

Comment:
If you already have the seed and the public key, then you already have the components. Thus, this function isn't necessary in Ed25519KeyPairComponents.

Reply:
Actually, I need it, because keys from SSH key files are not always in PKCS#8, and I still need to get their components.

Reply:
Sorry, I don't understand. There's already a public `pub fn from_seed_and_public_key()` function in `Ed25519KeyPair` so you can just use it directly without using `Ed25519KeyPairComponents` at all, right?